### ISSUE TITLE
Queries to a collection fail with an empty response if the collection contains specific JSON data

### ISSUE THREAD

**Issue by OrKoN** [arangodb/arangodb#1315]
Hi,

We've found a problem that some queries fail if collections contain some specific JSON data. I don't know what exactly in the data causes the failure.

Steps to reproduce:

1. Create a new collection e.g. `test`
2. Import the following json file https://www.dropbox.com/s/roael6lv15gu9ip/data.json?dl=0 which contains a list of countries with translated names
3. Open the API tool and create a POST request to /_api/cursor
4. As the query param use the following:

```
{"query":"LET nodes = (FOR node IN @@collection FILTER (node.`@id` == @iri) RETURN node), includes = (FOR node IN @@collection RETURN node) RETURN {nodes: nodes, includes: includes}","bindVars":{"@collection":"test","iri":"1282028"}}
```

and run the query.

Actual results:
Response Code: 201, empty response body (0 length body is invalid JSON)

Expected results:
Response Code: 201, the response body contains nodes from the collection or at least valid JSON with empty `nodes` object.

Linux Ubuntu 14.04 x86_64

```

architecture: 64bit
build-date: 2015-04-11 11:29:15
configure:  './configure' 'CC=gcc' 'CXX=g++' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib' '--libexecdir=/usr/lib' '--localstatedir=/var' '--sharedstatedir=/usr/com' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--enable-all-in-one-libev' '--enable-all-in-one-v8' '--enable-all-in-one-etcd' '--enable-internal-go' '--disable-mruby'
env:  CC='gcc' CXX='g++'
icu-version: 52.1
libev-version: 4.11
maintainer-mode: false
openssl-version: OpenSSL 1.0.1f 6 Jan 2014
server-version: 2.5.2
sizeof int: 4
sizeof void*: 8
v8-version: 3.31.74.1

```

We use arangojs to query the database, the arangojs fails throwing a SyntaxError when it tries to parse an empty response from arrangodb. Despite this the data seems to be available via ArangoDB UI - i.e. one can browse the countries from the file.

**Comment by jsteemann** [comment]
I was able to reproduce this with 2.5.2. Interestingly the server returns a proper JSON when I use the devel version. Will try to figure out what causes this.

**Comment by jsteemann** [comment]
Found the reason: the response is assembled in the server correctly and then UTF-8 normalization of the response kicks in. This seems to expand the response string slightly so a target buffer cannot hold it. Normalization therefore returns with an error and no response string.

Will need to investigate how to properly fix this.

**Comment by jsteemann** [comment]
The issue has been fixed in 2.5 and devel now. The fix will be included in the next releases (e.g. 2.5.3).

I'd like to incorporate the JSON test data you linked into our test suite and would like to ask your permission for doing so. Are you ok with that?

**Comment by OrKoN** [comment]
Thanks for the quick fix. Sure, you can use the data that I provided. It's taken originally from http://sws.geonames.org/ and licensed under a Creative Commons Attribution 3.0 License

Could you estimate when the next release will happen? Or maybe if there is a way to get the fix before the release? or maybe do you know which part of the data exactly causes the problem so I can remove it temporarily? Thank you.

**Comment by jsteemann** [comment]
I think a 2.5.3 release can be built next week.

The problems seem to be caused by the `alternateName` attribute, specifically for the `@language` with value `bn`.

I tried copying the original data into a new collection with the `bn` language removed, and that seemed to work when running the original query. Here's my copy function for reference:
```js
var _ = require("underscore");
db._create("copy");
var docs = db.test.toArray();
for (var i = 0; i < docs.length; ++i) { 
  var doc = _.clone(docs[i]); 
  doc.alternateName = docs[i].alternateName.filter(function(orig) { 
    return orig["@language"] !== "bn"; 
  }); 
  db.copy.insert(doc); 
}
```

**Comment by jsteemann** [comment]
Should be fixed in 2.5.3.
